src/vectors.jl

```diff
@@ -43,24 +43,18 @@ Vec3f(1, 2, 3) # explicitly ask for single precision
 """
 struct Vec{Dim,T} <: StaticVector{Dim,T}
   coords::NTuple{Dim,T}
-  Vec{Dim,T}(coords::NTuple{Dim,T}) where {Dim,T} = new{Dim,T}(coords)
-  Vec{Dim,T}(coords::NTuple{Dim,T}) where {Dim,T<:Integer} = new{Dim,Float64}(coords)
+  Vec(coords::NTuple{Dim,T}) where {Dim,T} = new{Dim,float(T)}(NTuple{Dim,float(T)}(coords))
```

Comment:
Can we add type parameters for the inner constructor? I think it is the default, we never used inner constructors in Julia without these parameters.

Reply:
The previous inner constructors presume that the type parameter of `Vec` is the same as that of `coords`.
Instead, my code infers the type parameter of `Vec`  from `coords`.
The proposed code will work like the previous one with more relaxed conditions.